extensions/sql-migration/src/dialog/migrationCutover/migrationCutoverDialog.ts

```diff
@@ -375,9 +384,12 @@ export class MigrationCutoverDialog {
 			await this.refreshStatus();
 		}));
 
-		headerActions.addItem(this._cutoverButton, {
-			flex: '0'
-		});
+		if (this._isProvisioned() && this._isOnlineMigration()) {
+			// _todo: should we just not create the cutover button if offline?
+			headerActions.addItem(this._cutoverButton, {
```

Comment:
I agree it would be good to only create the controls that are needed

Reply:
I wanted to create the necessary controls only, but I realized that there are cases where we can't properly determine if it's necessary or not.

While the migration is being created (provisioningStatus=Creating), we can't determine if the migration is online or offline.
For the cutover button, I decided to hide it by default. One we can determine the migration mode, the button will show up or remain hidden.